### PR TITLE
[PyAPI][Opset13] ScaledDotProductAttention-13 PyAPI improvements + tests

### PR DESCRIPTION
**Pull request by mmikolajcz**
### Details:
 - *Fix issues with ScaledDotProductAttention PyAPI*
 - *Add missing tests*

### Tickets:
 - *126324*

### CODE REVIEW

File: src/bindings/python/tests/test_graph/test_scaled_dot_product_attention.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+
+import openvino.runtime.opset13 as ops
+from openvino.runtime import PartialShape, Type
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_scaled_dot_product_attention.py:9]:
```suggestion
from openvino import PartialShape, Type
```

File: src/bindings/python/src/openvino/runtime/opset13/ops.py [hunk 2]
```diff
@@ -255,8 +257,13 @@ def scaled_dot_product_attention(
 
     :return: The new node performing Scaled Dot Product Attention operation.
     """
-    inputs = as_nodes(query, key, value, attention_mask) if attention_mask is not None else as_nodes(
-        query, key, value, scale)
+    inputs = as_nodes(query, key, value)
+    if attention_mask is not None:
+        inputs.append(as_node(attention_mask))
+    elif scale is not None:
+        inputs.append(as_node(convert_like(constant(0, Type.i32), inputs[0])))
```
[the remark this answers is not in the document]
Comment (mmikolajcz) [reply src/bindings/python/src/openvino/runtime/opset13/ops.py]: Thanks @mitruska, applied suggestion

**Review by jiwaszki — COMMENTED** [review]
LGTM from PyOV side

**Review by mitruska — APPROVED** [review]

File: src/bindings/python/src/openvino/runtime/opset13/ops.py [hunk 3]
```diff
@@ -255,8 +257,13 @@ def scaled_dot_product_attention(
...
     """
-    inputs = as_nodes(query, key, value, attention_mask) if attention_mask is not None else as_nodes(
-        query, key, value, scale)
+    inputs = as_nodes(query, key, value)
+    if attention_mask is not None:
+        inputs.append(as_node(attention_mask))
+    elif scale is not None:
+        inputs.append(as_node(convert_like(constant(np.array(0, np.int32)), inputs[0])))
+    if scale is not None:
+        inputs.append(as_node(scale))
```
Comment (PiotrKrzem) [inline src/bindings/python/src/openvino/runtime/opset13/ops.py:266]: What is the difference between setting the mask to 0 when scale is not None and having mask = None when scale is None?

Comment (mmikolajcz) [reply src/bindings/python/src/openvino/runtime/opset13/ops.py:266]: If I understand your question correctly, it is just for user experience to allow to use optional inputs like in C++ API. So when:

- both attn and scale is None, it will just pass query, key and value inputs to C++ constructor; 
- attn is provided and scale is None - pass query, key, value and attention;
- attn is None and scale is provided - fill attn with const 0 for C++ constructor and pass query, key, value, attn, scale;
- both attn and scale provided - pass all inputs
